### PR TITLE
Turn on -preview=markdown

### PR DESCRIPTION
**Pull request by dgileadi**
This PR does two things:

1. Adds the `-preview=markdown` flag to the Phobos Ddoc build.
2. Fixes places in the docs that were unintentionally interpreted as markdown.

The fixes were generally for:

* Lines that start with `-`, `+` or `*`, which in Markdown starts a list item.
* Backslashes that aren't in code blocks, and so are treated as escaping the next character.
* Words in brackets that aren't in code blocks, which Ddoc treats as [reference links](https://dlang.org/spec/ddoc.html#reference_links) to symbols with the same name.

### CODE REVIEW

File: etc/c/sqlite3.d [hunk 1]
```diff
@@ -1,35 +1,35 @@
...
+* if experience from use "in the wild" suggest such changes are prudent.
+*
+* The official C-language API documentation for SQLite is derived
+* from comments in this file.  This file is the authoritative source
+* on how SQLite interfaces are suppose to operate.
+*
+* The name of this file under configuration management is "sqlite.h.in".
+* The makefile makes some minor changes to this file (such as inserting
+* the version number) and changes its name to "sqlite3.h" as
+* part of the build process.
```
Comment (CyberShadow) [inline etc/c/sqlite3.d]: This is a regular comment, not a doc comment. Why are these changes necessary?

File: etc/c/sqlite3.d [hunk 2]
```diff
@@ -2094,49 +2094,49 @@ enum
...
+* 2010 August 30
+*
+* The author disclaims copyright to this source code.  In place of
+* a legal notice, here is a blessing:
+*
+*    May you do good and not evil.
+*    May you find forgiveness for yourself and forgive others.
+*    May you share freely, never taking more than you give.
+*
+************************************************************************
```
Comment (CyberShadow) [inline etc/c/sqlite3.d]: Ditto (and more in the same file)

File: std/regex/package.d [hunk 3]
```diff
@@ -136,7 +136,7 @@ $(TR $(TD Objects) $(TD
     $(REG_ROW \W, Matches any non-word character.)
     $(REG_ROW \s, Matches whitespace, same as \p{White_Space}.)
     $(REG_ROW \S, Matches any character except those recognized as $(I \s ). )
-    $(REG_ROW \\, Matches \ character. )
+    $(REG_ROW \\\\, Matches \ character. )
```
Comment (CyberShadow) [inline std/regex/package.d]: It would have been nice to make it so that it would be possible to write DDoc that would render the same with or without `-preview=markdown`, but I guess such changes are unavoidable in practice. Main concern is tools which consume DDoc directly, such as DDox, or editors/IDEs which parse DDoc to display formatted documentation in their UI.

File: etc/c/sqlite3.d (hunk 1 above)
Comment (dgileadi) [reply etc/c/sqlite3.d]: Unnecessary—fixed.

**Comment by 12345swordy** [comment]
ping 3 days are up.
